### PR TITLE
Add kubernetes pods tutorial using playground

### LINKED ISSUE #9870: [Docs] Tutorial: Exploring Kubernetes Pods with Meshery Playground
_This child issue is related to "[Playground] Learning Paths: Cloud Native tutorials with Meshery Playground #9832". See this issue for more context._

### Current State

Meshery Docs are in need of new tutorials with hands-on labs. In this tutorial, we will explore how to use Meshery Playground, an interactive live cluster environment, to perform hands-on labs for managing Kubernetes deployments. Meshery is a powerful tool that simplifies the adoption and operation of cloud and cloud native technologies. Meshery Playground provides a safe and isolated environment for learning and experimenting with these technologies.

### Desired State

1. Take the following sample and refactor in context of Meshery Playground. 
1. Instead of `kubectl` commands use Meshery UI and MeshMap as the primary mechanisms for users to perform tasks in the labs. Put CLI commands inside of `<details><summary></summary></details>` as an alternative reference.
1. Reference existing Meshery Designs from the Meshery Catalog both as starting point and/or as a finished example of the lab.
1. As needed, take screenshots or video recordings to enrich the content.

```yaml
---
layout: default
title: Exploring Kubernetes Pods with Meshery Playground
abstract: Learn Kubernetes Pods with Meshery Playground
permalink: guides/tutorials/kubernetes-pods
redirect_from: guides/tutorials/kubernetes-pods/
type: guide
category: tutorial
language: en
abstract: "In this tutorial, we will explore Kubernetes Pods using Meshery Playground, an interactive live cluster environment, to perform hands-on labs for deploying and managing containerized applications at the Pod level."
---

Introduction:
In this tutorial, we'll dive into the fundamentals of Kubernetes Pods, the smallest deployable units in the Kubernetes ecosystem. Using Meshery Playground, an interactive live cluster environment, we'll conduct hands-on labs to gain practical experience in deploying, managing, and understanding the concepts related to Kubernetes Pods.

Prerequisites:
- Basic understanding of containerization and Kubernetes concepts.
- Meshery Playground access. If you don't have an account, sign up at [Meshery Playground](https://meshery.layer5.io/play).

Lab Scenario: Deploying and Exploring Pods in a Microservices Architecture

Objective:
Learn how to create, manage, and explore Kubernetes Pods within the context of a microservices architecture.

### Steps:

#### 1. **Accessing Meshery Playground:**
   - Log in to the [Meshery Playground](https://meshery.layer5.io/play) using your credentials.
   - Navigate to the Meshery Playground dashboard.

#### 2. **Creating a Simple Pod:**
   - Deploy a basic Pod with a single container using a YAML configuration file.

\```yaml
# simple-pod.yaml
apiVersion: v1
kind: Pod
metadata:
  name: mypod
spec:
  containers:
  - name: mycontainer
    image: nginx:latest
\```

Apply the Pod configuration:

\```bash
kubectl apply -f simple-pod.yaml
\```

#### 3. **Exploring Pod States and Information:**
   - Use Meshery Playground to explore the different states a Pod can be in and gather information about the deployed Pod.

\```bash
kubectl get pod mypod
kubectl describe pod mypod
\```

#### 4. **Interacting with Pods:**
   - Gain hands-on experience by interacting with the Pod using commands and Meshery Playground's interactive features.

\```bash
kubectl exec -it mypod -- /bin/bash
\```

#### 5. **Pod Networking:**
   - Explore networking aspects related to Pods, including accessing a Pod from another Pod.

\```bash
kubectl expose pod mypod --type=NodePort --port=80
\```

#### 6. **Deleting and Recreating Pods:**
   - Understand the impact of deleting and recreating Pods and observe how Meshery Playground reflects these changes.

\```bash
kubectl delete pod mypod
\```

#### 7. **Pod Annotations and Labels:**
   - Learn how to add annotations and labels to Pods and leverage them for better organization and management.

\```yaml
# annotated-labeled-pod.yaml
apiVersion: v1
kind: Pod
metadata:
  name: mypod
  annotations:
    environment: production
  labels:
    app: myapp
spec:
  containers:
  - name: mycontainer
    image: nginx:latest
\```

Apply the annotated and labeled Pod configuration:

\```bash
kubectl apply -f annotated-labeled-pod.yaml
\```

#### 8. **Clean-Up:**
   - Delete the Pods and associated resources after completing the lab.

\```bash
kubectl delete pod mypod
\```

#### 9. **Saving and Sharing:**
   - Save your scenario in Meshery Playground for future reference.
   - Share your Pod scenarios with the Meshery community for collaborative learning.

### Conclusion
Congratulations! You've successfully completed the lab on exploring Kubernetes Pods using Meshery Playground. This hands-on experience has provided valuable insights into the deployment, management, and interaction with Pods in a Kubernetes environment. Continue exploring more scenarios in the Meshery Playground to enhance your skills in container orchestration.

{% include related-content.html type="tutorials" %}



```

---

### Contributor [Guides](https://docs.meshery.io/project/contributing) and [Handbook](https://layer5.io/community/handbook)
- 📚 [Instructions for contributing to documentation](https://docs.meshery.io/project/contributing/contributing-docs)
   - Meshery documentation [site](https://docs.meshery.io/) and [source](https://github.com/meshery/meshery/tree/master/docs)
- 🛠 [Meshery Build & Release Strategy](https://docs.meshery.io/project/contributing/build-and-release)
- 🎨 Wireframes and [designs for Meshery UI](https://www.figma.com/file/SMP3zxOjZztdOLtgN4dS2W/Meshery-UI) in Figma [(open invite)](https://www.figma.com/team_invite/redeem/qJy1c95qirjgWQODApilR9)
- 🙋🏾🙋🏼 Questions: [Discussion Forum](http://discuss.meshery.io) and [Community Slack](https://slack.meshery.io)

### PR DESCRIPTION
**Pull request by sangramrath**
**Notes for Reviewers**

This PR fixes #9870 
- _Pod Networking_ is left out as terminal is not working, and its not possible to demonstrate network connectivity

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

**Comment by l5io** [comment]
🚀 Preview for commit 486a3c6b8029b317db2c829f7a5eafb9d9e6aa53 at: https://65e0d44f8aa49f16259779a8--meshery-docs-preview.netlify.app

**Comment by l5io** [comment]
🚀 Preview for commit 69abc326a9ac788235e1e90fd7da2ff8fa496b43 at: https://65e5d71d627e7925e5c998d2--meshery-docs-preview.netlify.app

**Review by sandramsc — COMMENTED** [review]
How does one access the tutorial? I have added an example of the path that you can see in the screenshot, I propose that your tutorial is add to the Tutorial tab.

**This tutorial path:**
MESH


**Suggested path:**
MESHH



[Link](https://github.com/meshery/meshery/pull/10107/files#diff-a15270653e60da41213aebc78a8e96a376ca6ed954f0aa9c42f8ef40e1231e8b) to created navigation for your consideration.

**Comment by l5io** [comment]
🚀 Preview for commit 125e0d12c00eee503e34016dbee6ab052d6eef00 at: https://65e6aa27985e054fc6617ab9--meshery-docs-preview.netlify.app

**Comment by sangramrath** [comment]
The doc link is https://65e6aa27985e054fc6617ab9--meshery-docs-preview.netlify.app/guides/tutorials/kubernetes-pods. As far as I know the _tutorials_ option in the navigation menu (toc) isn't currently part of the master branch. When the PR with the navigation change gets pushed this should get resolved.

**Comment by l5io** [comment]
🚀 Preview for commit 3ce01ffa310072e5d8fdc76fe486ace465f053ce at: https://65e7e86f76a224af3c8fd138--meshery-docs-preview.netlify.app

File: docs/pages/guides/tutorials/kubernetes-pods.md [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+      ![](./screenshots/2024-02-22_18-35.png)
+   
+#### Add a container to the Pod
+The pod at this stage does not have a container, so we will add one. This is similar to defining the containers in the _spec:_ section of a YAML file. For this hands-on we will make some basic and necessary configurations only.
+
+   7. Scroll down within the Pod configuration modal to the Containers section. Click **+** to add a container.
+      ![](./screenshots/2024-02-23_11-54.png)
+   8. Expand **Containers-0**.
+      ![](./screenshots/2024-02-23_11-54_1.png)
+   9. Next, fillout some of the requred container specifications. Start by entering the container image, we will use _nginx:latest_ for this exercise.
```
Comment (vishalvivekm) [inline docs/pages/guides/tutorials/kubernetes-pods.md:57]:
```suggestion
   9. Next, fill out some of the required container specifications. Start by entering the container image, we will use _nginx:latest_ for this exercise.
```

**Comment by l5io** [comment]
🚀 Preview for commit e4907daee979a1dd3346eea3da94d6c9c461dbf2 at: https://65e803a518b65ad1ca89e8e5--meshery-docs-preview.netlify.app

**Comment by vishalvivekm** [comment]
As https://github.com/meshery/meshery/pull/9950 is still a WIP, I've manually added this `Tutorial` to the left TOC.

**Comment by l5io** [comment]
🚀 Preview for commit 766e06e3e713ec7835b695ddaf5373471a4107aa at: https://65e804e476a224c2878fd1ef--meshery-docs-preview.netlify.app

**Review by vishalvivekm — APPROVED** [review]
LGTM.
Thank you @sangramrath for getting the first Tutorial out :) 🚀

**Review by MUzairS15 — APPROVED** [review]
Looks great @sangramrath 
As an update, the `Terminal Session` and `Log Streamer` were recently fixed.
Can you please try and update the tutorial may be in this / a separate PR?

**Comment by vishalvivekm** [comment]
@sangramrath 🎈

**Pull request merged**